named_arrays/_vectors/vectors.py

```diff
@@ -85,6 +81,39 @@ def type_matrix(self) -> Type[na.AbstractExplicitMatrixArray]:
         The corresponding :class:`named_arrays.AbstractMatrixArray` class
         """
 
+    @property
+    def matrix(self) -> na.AbstractMatrixArray:
+        new_dict = {}
+        for c in self.components:
+            component = self.components[c]
+            if isinstance(component, AbstractVectorArray):
+                new_dict[c] = component.matrix
+            elif isinstance(component, na.AbstractMatrixArray):
+                return NotImplemented
```

Comment:
It would be better if this raised an error instead of returning `NotImplemented`. `NotImplemented` is intended to be used by a function that will raise an error if all the implementations return `NotImplemented`.

If you return `NotImplemented` in this way, it'll act like NaN and poison all downstream results without raising an error.